test/FunctionsV2/CorrelationEndToEndTests.cs

```diff
@@ -155,6 +155,62 @@ public async Task AllOrchestrationActivityActions(bool extendedSessions, string
             return new Tuple<List<OperationTelemetry>, List<ExceptionTelemetry>>(result.CorrelationSort(), exceptionTelemetryList);
         }
 
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void TelemetryClientSetupWithoutAppInsightsInstrumentationKeyEmitsWarning(bool extendedSessions)
+        {
+            TraceOptions traceOptions = new TraceOptions()
+            {
+                DistributedTracingEnabled = true,
+                DistributedTracingProtocol = "W3CTraceContext",
+            };
+
+            DurableTaskOptions options = new DurableTaskOptions();
+            options.Tracing = traceOptions;
+
+            using (var host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                "SingleOrchestration",
+                extendedSessions,
+                options: options))
+            {
+                string warningMessage = "'APPINSIGHTS_INSTRUMENTATIONKEY' isn't defined in the current environment variables, but Distributed Tracing is enabled. Please set 'APPINSIGHTS_INSTRUMENTATIONKEY' to use Distributed Tracing.";
+                var warningLogMessage = this.loggerProvider.GetAllLogMessages().ToList().FindAll(l => l.FormattedMessage.Equals(warningMessage));
+                Assert.True(warningLogMessage.Count != 0);
```

Comment:
I think xunit should normally issue warnings about this, but you should generally avoid using `Assert.True(...)` for things like this and instead use more specific asserts, like `Assert.NotEqual(0, warningLogMessage.Count)` or `Assert.NotEmpty(warningLogMessage)`. One of the benefits is that you get better error messages if the check fails. If you want to get even fancier, you could use:
```csharp
Assert.Single(this.loggerProvider.GetAllLogMessages().Where(l => l.FormatMessage.Equals(warningMessage));
```